src/serializer/ResidualHeapSerializer.js

```diff
@@ -1200,7 +1200,7 @@ export class ResidualHeapSerializer {
     // add strict modes
     let strictDirective = t.directive(t.directiveLiteral("use strict"));
     let globalDirectives = [];
-    if (!unstrictFunctionBodies.length && strictFunctionBodies.length) {
+    if (!this.realm.isStrict && !unstrictFunctionBodies.length && strictFunctionBodies.length) {
       // no unstrict functions, only strict ones
       globalDirectives.push(strictDirective);
```

Comment:
I don't think so. If the realm is strict then the directive goes to the start of the global code and there is no need to repeat it inside the wrapper.